### ISSUE TITLE
refactor: the Command struct's `Exec()` should return (ExitStatus, err)

### ISSUE THREAD

**Issue by jmontroy90** [Praqma/helmsman#598]
Code that uses `ExitStatus` returned from some `status := cmd.Exec()` typically conditions on `status.code != 0`. Then `status.errors` is included in the subsequent error.

This frequently leads to information being dropped:
- Not every CLI displays anything other than a non-zero exit code for errors (often the error code itself has semantic meaning)
- Not every CLI we might use dumps errors to stderr, instead dumping to stdout
- Sometimes more relevant info for debugging is on stdout
- Slightly different: we're often constructing `ExitStatus` directly from application failures instead of invoked cmd failures, conflating the two failure modes.

As such, I propose the following refactor:

1. `cmd.Exec()` returns `(ExitStatus, err)`.
2. `ExitStatus` would be roughly the same struct, and would have a `String()` that can also be called that will include any relevant CLI stuff (stdout, stderr)
3. The returned `err` would just be some `FailureStatus` struct that implements `Error()`, where that implementation would include stdout, stderr, and the return code

The benefit of this is that we'd be able to have pretty idiomatic, consistent code everywhere, e.g.:
```
out, err := myCmd.exec()
if err != nil {
    return fmt.Errorf("error in my special command: %w", err)
}

return fmt.Sprintf("my cool cmd results: %s", out)
```

I've got a start on this already, but wanted to get some input before I committed.

**Comment by luisdavim** [comment]
Sounds good to me but I don't see a reason for the error to be some custom structure, it can simply be a regular error and we'd still return the ExitStatus even on errors and carry the information there....

**Comment by luisdavim** [comment]
fixed in #601
